src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/DataBaseHandler.java

```diff
@@ -0,0 +1,231 @@
+        }
+    }
+
+    /**
+     * This function loops the values in the Table enum and executes the create
+     * table command for each table and creates it in the database if it not already
+     * exist.
+     *
+     * @throws ConnectException
+     */
+    private void createTables() throws ConnectException {
+        try (Connection connection = connect(); Statement statement = connection.createStatement()) {
+            for (Table table : Table.values()) {
+                createTable(table, statement);
+            }
+            LOGGER.debug("Created tables successfully");
+        } catch (SQLException e) {
+            LOGGER.error("Error while creating Tables in database: {}\n{}", e.getMessage(), e);
+        }
+    }
+
+    /**
+     * This function connects to the database and returns a connection
+     *
+     * @return Connection to database
+     * @throws ConnectException
+     */
+    private Connection connect() throws ConnectException {
+        try {
+            Class.forName("org.sqlite.JDBC");
+            return DriverManager.getConnection(this.databaseFile);
+        } catch (SQLException | ClassNotFoundException e) {
+            throw new ConnectException(
+                    String.format("Failed to create a database connection. %s\n%s", e.getMessage(), e));
+        }
+    }
+
+    /**
+     * Executes the prepared statement.
+     *
+     * @param preparedStatement
+     * @return
+     * @throws SQLException
+     */
+    private String executeQuery(final PreparedStatement preparedStatement) throws SQLException {
+        try (ResultSet result = preparedStatement.executeQuery();) {
+            if (result.next()) {
+                return result.getString(EVENT_ID_KEY);
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Creates a prepared statement and executes an update on the sqlStatement
+     * given.
+     *
+     * @param sqlStatement
+     * @param keyValue
+     * @param eiffelEvent
+     * @throws ConnectException
+     * @throws SQLException
+     */
+    private void executeUpdate(final String sqlStatement, final String keyValue, final String eiffelEvent)
+            throws ConnectException, SQLException {
+        try (PreparedStatement preparedStatement = prepareStatmentForResourceBlock(sqlStatement)) {
+            preparedStatement.setString(1, eiffelEvent);
+            preparedStatement.setString(2, keyValue);
+            int updateCount = preparedStatement.executeUpdate();
+
+            if (updateCount == 0) {
+                throw new SQLException("No changes was saved in the database.");
+            }
+
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to add value into database: {}\n{}", e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    public void createTable(final Table table, final Statement statement) throws ConnectException, SQLException {
```

Comment:
Why public?

Reply:
I have no idéa, will change in upcoming change